js_modules/dagit/packages/core/src/assets/__tests__/AssetView.test.tsx

```diff
@@ -1,74 +1,111 @@
-import {act, render, screen} from '@testing-library/react';
-import faker from 'faker';
+import {MockedProvider} from '@apollo/client/testing';
+import {act, render, screen, waitFor} from '@testing-library/react';
 import * as React from 'react';
 import {MemoryRouter} from 'react-router-dom';
 
-import {TestProvider} from '../../testing/TestProvider';
+import {AssetKeyInput} from '../../graphql/types';
 import {AssetView} from '../AssetView';
+import {
+  AssetViewDefinitionSourceAsset,
+  AssetViewDefinitionNonSDA,
+  LatestMaterializationTimestamp,
+  AssetViewDefinitionSDA,
+  AssetGraphEmpty,
+} from '../__fixtures__/AssetViewDefinition.mocks';
 
 // This file must be mocked because Jest can't handle `import.meta.url`.
 jest.mock('../../graph/asyncGraphLayout', () => ({}));
 
-// This file must be mocked because useVirtualizer tries to create a ResizeObserver,
+// These files must be mocked because useVirtualizer tries to create a ResizeObserver,
 // and the component tree fails to mount.
 jest.mock('../AssetPartitions', () => ({AssetPartitions: () => <div />}));
+jest.mock('../AssetEvents', () => ({AssetEvents: () => <div />}));
 
 describe('AssetView', () => {
-  const mocks = {
-    AssetNode: () => ({
-      partitionKeys: () => [...new Array(0)],
-      partitionKeysByDimension: () => [...new Array(1)],
-    }),
-    Asset: () => ({
-      assetMaterializations: () => [...new Array(1)],
-      assetObservations: () => [...new Array(0)],
-    }),
-    MaterializationEvent: () => ({
-      timestamp: () => '100',
-    }),
-    MetadataEntry: () => ({
-      __typename: 'TextMetadataEntry',
-      label: () => faker.random.word().toLocaleLowerCase(),
-    }),
-  };
-
-  const Test = ({path}: {path: string}) => {
+  const Test = ({path, assetKey}: {path: string; assetKey: AssetKeyInput}) => {
     return (
-      <TestProvider apolloProps={{mocks}}>
+      <MockedProvider
+        mocks={[
+          AssetGraphEmpty,
+          AssetViewDefinitionSDA,
+          AssetViewDefinitionNonSDA,
+          AssetViewDefinitionSourceAsset,
+        ]}
+      >
         <MemoryRouter initialEntries={[path]}>
-          <AssetView assetKey={{path: ['foo']}} />
+          <AssetView assetKey={assetKey} />
         </MemoryRouter>
-      </TestProvider>
+      </MockedProvider>
     );
   };
 
   const MESSAGE = /this is a historical view of materializations as of \./i;
 
+  describe('Launch button', () => {
+    it('shows the "Materialize" button for a software-defined asset', async () => {
+      await act(async () => {
+        render(<Test path="/sda_asset" assetKey={{path: ['sda_asset']}} />);
+      });
```

Comment:
Is it still necessary to async-render this, now that it's a `MockedProvider` instead of a `TestProvider`?